pkg/api/trillian_client.go

```diff
@@ -76,6 +77,52 @@ func (t *TrillianClient) addLeaf(byteValue []byte) *Response {
 	}
 	resp, err := t.client.QueueLeaf(t.context, rqst)
 
+	// check for error
+	if err != nil || (resp.QueuedLeaf.Status != nil && resp.QueuedLeaf.Status.Code != int32(codes.OK)) {
+		return &Response{
+			status:       status.Code(err),
+			err:          err,
+			getAddResult: resp,
+		}
+	}
+
+	root, err := t.root()
+	if err != nil {
+		return &Response{
+			status:       status.Code(err),
+			err:          err,
+			getAddResult: resp,
+		}
+	}
+	verifier, err := client.NewLogVerifierFromTree(t.tree)
+	if err != nil {
+		return &Response{
+			status:       status.Code(err),
+			err:          err,
+			getAddResult: resp,
+		}
+	}
+	logClient := client.New(t.logID, t.client, verifier, root)
+	if err := logClient.WaitForInclusion(t.context, byteValue); err != nil {
```

Comment:
Is there a way to set a timeout here? We don't have a way to expose metrics yet, but this time span is exactly what I would want to monitor eventually too.

Reply:
no, it will be min({HTTP request timeout (as set by client) which will cause context to cancel},{10 seconds}). https://github.com/google/trillian/blob/549baa4214f1fd64b89344d55dc9aa894ab80c27/client/log_client.go#L130 uses an backoff approach here. I suspect that if we add metrics we can just wrap the call and be good with it.